api_core/google/api_core/grpc_helpers.py

```diff
@@ -136,3 +138,178 @@ def create_channel(target, credentials=None, scopes=None, **kwargs):
 
     return google.auth.transport.grpc.secure_authorized_channel(
         credentials, request, target, **kwargs)
+
+
+_ChannelCall = collections.namedtuple(
+    '_ChannelCall', ['request', 'timeout', 'metadata', 'credentials'])
+
+
+class _CallableStub(object):
+    """Stub for the grpc.*MultiCallable interfaces."""
+
+    def __init__(self, method, channel):
+        self._method = method
+        self._channel = channel
+        self.response = None
+        """Union[protobuf.Message, Callable[protobuf.Message], exception]:
+        The response to give when invoking this callable. If this is a
+        callable, it will be invoked with the request protobuf. If it's an
+        exception, the exception will be raised when this is invoked.
+        """
+        self.responses = None
```

Comment:
How does this interact with a unary-stream call? Or a stream-stream, for that matter?
("Out of scope" is a valid answer.)

Reply:
For stream responses, you're expected to set `response` to an iterable:

```python
channel.SomeStreamingThing.response = [resp1, resp2, resp3]
```

For multiple invocations of the same streaming method, you should set `responses` to an iterable of iterables:

```python
channel.SomeStreamingThing.responses = [
    [resp1, resp2],
    [resp3, resp4]
]
```

This is the reason for the `response`/`responses` split.